yt/yt/server/job_proxy/user_job.cpp

```diff
@@ -641,57 +641,71 @@ class TUserJob
+                auto portoJobEnvironmentConfig = ConvertTo<TPortoJobEnvironmentConfigPtr>(Config_->JobEnvironment);
+                auto portoExecutor = NContainers::CreatePortoExecutor(portoJobEnvironmentConfig->PortoExecutor, "job-shell");
+
+                std::vector<TString> shellEnvironment;
+                shellEnvironment.reserve(Environment_.size());
+                std::vector<TString> visibleEnvironment;
+                visibleEnvironment.reserve(Environment_.size());
+
+                for (const auto& variable : Environment_) {
+                    if (variable.StartsWith(NControllerAgent::SecureVaultEnvPrefix) &&
+                        !UserJobSpec_.enable_secure_vault_variables_in_job_shell())
+                    {
+                        continue;
+                    }
+                    if (variable.StartsWith("YT_")) {
+                        shellEnvironment.push_back(variable);
+                    }
+                    visibleEnvironment.push_back(variable);
                 }
-                if (variable.StartsWith("YT_")) {
-                    shellEnvironment.push_back(variable);
+
+                auto shellManagerUid = UserId_;
+                if (Config_->TestPollJobShell) {
+                    shellManagerUid = std::nullopt;
+                    shellEnvironment.push_back("PS1=\"test_job@shell:\\W$ \"");
                 }
-                visibleEnvironment.push_back(variable);
-            }
 
-            auto shellManagerUid = UserId_;
-            if (Config_->TestPollJobShell) {
-                shellManagerUid = std::nullopt;
-                shellEnvironment.push_back("PS1=\"test_job@shell:\\W$ \"");
-            }
+                std::optional<int> shellManagerGid;
+                // YT-13790.
+                if (Config_->RootPath) {
+                    shellManagerGid = 1001;
+                }
 
-            std::optional<int> shellManagerGid;
-            // YT-13790.
-            if (Config_->RootPath) {
-                shellManagerGid = 1001;
+                    // ToDo(psushin): move ShellManager into user job environment.
+                    TShellManagerConfig config{
+                        .PreparationDir = Host_->GetPreparationPath(),
+                        .WorkingDir = Host_->GetSlotPath(),
+                        .UserId = shellManagerUid,
+                        .GroupId = shellManagerGid,
+                        .MessageOfTheDay = Format("Job environment:\n%v\n", JoinToString(visibleEnvironment, TStringBuf("\n"))),
+                        .Environment = std::move(shellEnvironment),
+                        .EnableJobShellSeccopm = Config_->EnableJobShellSeccopm,
+                    };
+
+                    ShellManager_ = CreatePortoShellManager(
+                        config,
+                        portoExecutor,
+                        UserJobEnvironment_->GetUserJobInstance());
+                    break;
+                }
+            default: {
```

Comment:
In situation of all cases presented default case is not needed

Reply:
Yes, but this default can help in case of new enum value added. I think that default with crash is a common practice in our project.

Reply:
You will receive a compiler error if you try to add a new value to enum (in case when default section is removed)